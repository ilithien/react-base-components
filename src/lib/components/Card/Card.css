.react-base__card {
  -webkit-box-shadow: 5px 5px 10px -5px rgba(0,0,0,0.75);
  -moz-box-shadow: 5px 5px 10px -5px rgba(0,0,0,0.75);
  box-shadow: 5px 5px 10px -5px rgba(0,0,0,0.75);    
  margin: 10px;
  border: 1px solid #ccc;  
}

.react-base__card > img {
  width: 100%;
}

.react-base__card-content {
  padding: 5px 5px 10px 20px;
}