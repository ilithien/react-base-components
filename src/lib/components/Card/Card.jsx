import React from 'react'
import PropTypes from 'prop-types'
import './Card.css';

const Card = ({ image, title, description }) => {
  return (
    <div className="react-base__card">
      <img src={image} alt={title} title={title} />
      <div className="react-base__card-content">
      <h1>{title}</h1>
      <p>{description}</p>
      </div>
    </div>
  )
}

Card.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string
}

export default Card

