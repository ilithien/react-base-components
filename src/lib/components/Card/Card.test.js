import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";

describe("Testing Card component", () => {
  it("matches snapshot", () => {
    const snap = renderer.create(
      <Card />
    );

    const tree = snap.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
