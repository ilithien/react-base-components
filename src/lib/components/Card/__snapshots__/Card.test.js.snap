// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component matches snapshot 1`] = `
<div
  className="react-base__card"
>
  <img
    alt={undefined}
    src={undefined}
    title={undefined}
  />
  <div
    className="react-base__card-content"
  >
    <h1 />
    <p />
  </div>
</div>
`;
